File: check_table_structure.py
#!/usr/bin/env python3
"""
Check table structure to understand the correct column names
"""

import psycopg2

def check_table_structure():
    """Check the structure of the customermaster table"""
    print("🔍 Checking customermaster table structure...")
    
    try:
        conn = psycopg2.connect(
            host='localhost',
            port=5432,
            database='soft_sme_db',
            user='ai_assistant',
            password='ai_secure_password_2024'
        )
        
        cursor = conn.cursor()
        
        # Get column information
        cursor.execute("""
            SELECT column_name, data_type, is_nullable
            FROM information_schema.columns
            WHERE table_name = 'customermaster'
            ORDER BY ordinal_position
        """)
        
        columns = cursor.fetchall()
        
        print("📋 customermaster table structure:")
        print("=" * 50)
        for column_name, data_type, is_nullable in columns:
            nullable = "NULL" if is_nullable == "YES" else "NOT NULL"
            print(f"   {column_name}: {data_type} ({nullable})")
        
        # Get a sample row
        cursor.execute("SELECT * FROM customermaster LIMIT 1")
        sample = cursor.fetchone()
        
        if sample:
            print(f"\n📝 Sample row:")
            print("=" * 30)
            for i, (column_name, data_type, is_nullable) in enumerate(columns):
                value = sample[i] if sample[i] is not None else "NULL"
                print(f"   {column_name}: {value}")
        
        cursor.close()
        conn.close()
        
    except Exception as e:
        print(f"❌ Error checking table structure: {e}")

if __name__ == "__main__":
    check_table_structure() 

File: soft-sme-backend/check_table_structure.js
const { Pool } = require('pg');

// Database connection
const pool = new Pool({
  user: 'postgres',
  host: 'localhost',
  database: 'soft_sme_db',
  password: '123',
  port: 5432,
});

async function checkTableStructure() {
  const client = await pool.connect();
  
  try {
    console.log('🔍 Checking table structure...\n');
    
    const tables = [
      'salesorderlineitems',
      'purchaselineitems', 
      'sales_order_parts_to_order',
      'aggregated_parts_to_order',
      'inventory_vendors'
    ];
    
    for (const tableName of tables) {
      console.log(`\n📋 Table: ${tableName}`);
      
      // Get column information
      const columns = await client.query(`
        SELECT column_name, data_type, is_nullable
        FROM information_schema.columns 
        WHERE table_name = $1 
        ORDER BY ordinal_position
      `, [tableName]);
      
      console.log('   Columns:');
      columns.rows.forEach(col => {
        console.log(`      - ${col.column_name} (${col.data_type}) ${col.is_nullable === 'YES' ? 'NULL' : 'NOT NULL'}`);
      });
      
      // Get primary key
      const pk = await client.query(`
        SELECT kcu.column_name
        FROM information_schema.table_constraints tc
        JOIN information_schema.key_column_usage kcu ON tc.constraint_name = kcu.constraint_name
        WHERE tc.table_name = $1 AND tc.constraint_type = 'PRIMARY KEY'
      `, [tableName]);
      
      if (pk.rows.length > 0) {
        console.log(`   Primary Key: ${pk.rows[0].column_name}`);
      } else {
        console.log(`   Primary Key: None found`);
      }
      
      // Check if part_id column exists
      const partIdExists = columns.rows.some(col => col.column_name === 'part_id');
      console.log(`   Has part_id: ${partIdExists ? '✅' : '❌'}`);
    }
    
  } catch (error) {
    console.error('❌ Check failed:', error);
  } finally {
    client.release();
    await pool.end();
  }
}

// Run the check
checkTableStructure().catch(console.error);
